src/nettests/index.js

```diff
@@ -1,9 +1,106 @@
+import camelCase from 'camelcase'
+
+import percentage from '../cli/output/percentage'
+import wait from '../cli/output/wait'
+
+import {
+  Result,
+  Measurement
+} from '../config/db'
+
 import HTTPHeaderFieldManipulation from './http-header-field-manipulation'
 import HTTPInvalidRequestLine from './http-invalid-request-line'
 import WebConnectivity from './web-connectivity'
 import NDT from './ndt'
 import Base from './base'
 
+export const makeOoni = () => {
+  let dbOperations = [],
+      progress = null,
+      reportId = null,
+      uploaded = false,
+      localReportId = null,
+      isMk = false
+
+  let measurement = Measurement.build({
+    state: 'active'
+  })
+
+  const init = (nt) => {
+    // XXX we use duck typing to see if the argument is a measurement Kit
+    // nettest. Maybe we should change this.
+    if (nt.test && nt.name && nt.setOptions) isMk = true
+
+    measurement.name = camelCase(nt.name)
+    measurement.reportFile = measurement.makeReportFile()
+    dbOperations.push(measurement.save())
+
+    if (isMk) {
+      nt.test.set_options('no_file_report', '0')
```

Comment:
This is the default. But I like that you're setting it nonetheless for robustness.